File: python3_solution_set2/2389. Longest Subsequence With Limited Sum.py
import bisect
class Solution:
    def answerQueries(self, nums: List[int], queries: List[int]) -> List[int]:
        
        ans = []
        
        nums.sort()
        pre = [0] + list(itertools.accumulate(nums))
        
        for x in queries:
            done = False
            for y in range(len(pre)):
                if pre[y] > x:
                    done = True
                    ans.append(y-1)
                    break
            if done == False:
                ans.append(len(nums))
        
        return ans
        

File: cpp_solutions/2389. Longest Subsequence With Limited Sum.cc
class Solution {
public:
    vector<int> answerQueries(vector<int>& nums, vector<int>& queries) {
        vector<int> ans;
        
        sort(nums.begin(),nums.end());
        for (auto x: queries) {
            int curr = 0;
            bool done = false;
            for(int y =0; y < nums.size();++y) {
                curr+=nums[y];
                // cout << curr << " ";
                if (curr > x) {
                    ans.emplace_back(y);
                    done = true;
                    break;
                }
            
            }
            if (done == false) {
                
                ans.emplace_back(nums.size());
            }
            // cout << "\n";
            
        }
        return ans;
        
    }
};
